tests/modules/images/webp-uploads/helper-tests.php

```diff
@@ -410,4 +410,52 @@ function () {
 		$this->assertIsArray( $transforms );
 		$this->assertSame( array( 'image/jpeg' => array( 'image/jpeg', 'image/webp' ) ), $transforms );
 	}
+
+	/**
+	 * Tests the webp_uploads_discard_larger_generated_images filter.
+	 *
+	 * @dataProvider data_provider_image_filesize
+	 *
+	 * @test
+	 */
+	public function it_should_check_if_the_additional_image_is_larger_than_the_original_image( $original_filesize, $additional_filesize, $expected_status ) {
+		add_filter( 'webp_uploads_discard_larger_generated_images', '__return_true' );
+
+		$output = webp_uploads_should_discard_additional_image_file( $original_filesize, $additional_filesize );
+		$this->assertSame( $output, $expected_status );
+	}
+
+	public function data_provider_image_filesize() {
+		return array(
+			array(
+				array( 'filesize' => 120101 ),
+				array( 'filesize' => 100101 ),
+				false,
+			),
+			array(
+				array( 'filesize' => 100101 ),
+				array( 'filesize' => 120101 ),
+				true,
+			),
+			array(
+				array( 'filesize' => 10101 ),
+				array( 'filesize' => 10101 ),
+				true,
+			),
+		);
+	}
+
+	/**
+	 * Check the disabled behaviour of the webp_uploads_discard_larger_generated_images filter.
+	 *
+	 * @dataProvider data_provider_image_filesize
+	 *
+	 * @test
+	 */
+	public function it_should_return_an_additional_image_if_it_is_bigger_than_original( $original_filesize, $additional_filesize ) {
```

Comment:
That method name is not really accurate, this method is about testing that no images are discarded regardless of file size if the filter is set to `false`.

Reply:
Thank you!